webserver/static/js/listens-viewer.js

```diff
@@ -17,18 +18,42 @@ var ListensViewer = React.createClass({
                           "in the data-user-id property.");
             return;
         }
-        $.get("/listen/user/" + container.dataset.userId, function(data) {
-            if (this.isMounted()) {
-                this.setState({
-                    userId: container.dataset.userId,
-                    listens: data.payload.listens
-                });
-            }
-        }.bind(this));
+        window.addEventListener('popstate', this.handleLoadData);
+        this.handleLoadData();
     },
     componentDidUpdate: function() {
         jQuery("abbr.timeago").timeago();
     },
+    handleLoadData: function () {
+        if (this.state.ignorePopState) { return; }
```

Comment:
That might be wrong though, since it could still have the `state` property of the added event, even if it's triggered by the user...

The other alternative I mentioned in IRC is just setting `this.ignorePopState = true/false` instead of using `setState`.